### PR TITLE
Use difficulty set in glowstone.yml

### PR DESCRIPTION
**Pull request by wow4201**
Allows user to change the server difficulty 0 (PEACEFUL),1(EASY),2(NORMAL),3(HARD) in glowstone.yml and take effect.
I personally wanted this implemented because 'Hunger now refills on Peaceful difficulty"

### CODE REVIEW

File: src/main/java/net/glowstone/GlowWorld.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ protected String disambiguate(World subject, String metadataKey) {
     private boolean autosave = true;
 
     /**
-     * The world's gameplay difficulty.
+     * The world's gameplay difficulty. PEACEFUL (0), EASY (1), NORMAL (2), HARD (3)
```
Comment (turt2live) [inline src/main/java/net/glowstone/GlowWorld.java]: The added documentation is not needed

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 2]
```diff
@@ -311,7 +311,7 @@ private boolean migrate() {
         // game props
         GAMEMODE("game.gamemode", "SURVIVAL", Migrate.PROPS, "gamemode"),
         FORCE_GAMEMODE("game.gamemode-force", "false", Migrate.PROPS, "force-gamemode"),
-        DIFFICULTY("game.difficulty", "EASY", Migrate.PROPS, "difficulty"),
+        DIFFICULTY("game.difficulty", 2, Migrate.PROPS, "difficulty"),
```
Comment (turt2live) [inline src/main/java/net/glowstone/util/ServerConfig.java:314]: Reason?

Comment (wow4201) [inline src/main/java/net/glowstone/util/ServerConfig.java:314]:
```
public Difficulty getDifficulty() {
    return Difficulty.getByValue(config.getInt(ServerConfig.Key.DIFFICULTY));
}
```

The reason is a string "EASY" isn't handled as well as the above code. Also this is what everyone is use to.

Comment (turt2live) [inline src/main/java/net/glowstone/util/ServerConfig.java:314]: I'm personally used to the string version. What's wrong with this?

```
try{
    // Can't remember if this throws an exception or not. It's too late in the night. 
    return Difficulty.valueOf(config.getString(ServerConfig.Key.DIFFICULTY).toUpperCase());
}catch(Exception e){
    // Return some default
}
```

File: src/main/java/net/glowstone/GlowServer.java [hunk 3]
```diff
@@ -1244,4 +1244,8 @@ public String getShutdownMessage() {
     public boolean getAllowFlight() {
         return config.getBoolean(ServerConfig.Key.ALLOW_FLIGHT);
     }
+
```
Comment (turt2live) [inline src/main/java/net/glowstone/GlowServer.java]: Whitespace

File: src/main/java/net/glowstone/GlowWorld.java [hunk 4]
```diff
@@ -235,7 +235,8 @@ public GlowWorld(GlowServer server, WorldCreator creator) {
         animalLimit = server.getAnimalSpawnLimit();
         waterAnimalLimit = server.getWaterAnimalSpawnLimit();
         ambientLimit = server.getAmbientSpawnLimit();
-
+        difficulty = server.getDifficulty();
+
```
Comment (turt2live) [inline src/main/java/net/glowstone/GlowWorld.java]: Whitespace

**Comment by turt2live** [comment]
@wow4201 this PR does not currently merge. Please fix this ASAP.

**Comment by SpaceManiac** [comment]
I notice your commits don't have an email attached. Is this intentional?

**Comment by Wild1145** [comment]
Whats with the removal of files like contributing,MD ?

File: build.gradle [hunk 5]
```diff
@@ -114,9 +107,15 @@ jar.manifest.mainAttributes(
...
         'Implementation-Title': description,
         'Implementation-Version': ext.gitDescribe,
-        'Implementation-Vendor': 'Glowstone Project',
-        'Specification-Title': 'Bukkit/Glowkit',
-        'Specification-Version': ext.bukkitVersion,
-        'Specification-Vendor': 'Glowstone Project',
+        'Implementation-Vendor': 'SpaceManiac',
+        'Specification-Title': 'Bukkit',
+        'Specification-Version': project.ext.bukkitVersion,
+        'Specification-Vendor': 'Bukkit Team',
```
Comment (Wild1145) [inline build.gradle:113]: This doesnt look right...

File: src/main/java/net/glowstone/GlowChunkSnapshot.java [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
 import org.bukkit.block.Biome;
 
+import net.glowstone.GlowChunk.ChunkSection;
+
 /**
  * Class representing a snapshot of a chunk.
  */
 public class GlowChunkSnapshot implements ChunkSnapshot {
-
+
```
Comment (Wild1145) [inline src/main/java/net/glowstone/GlowChunkSnapshot.java:14]: Unnecessary White space here.

File: src/main/java/net/glowstone/GlowChunkSnapshot.java [hunk 7]
```diff
@@ -164,7 +152,7 @@ private int coordToIndex(int x, int z) {
     }
 
     public static class EmptySnapshot extends GlowChunkSnapshot {
-
+
```
Comment (Wild1145) [inline src/main/java/net/glowstone/GlowChunkSnapshot.java:155]: Again, a line that didn't need change & now with unnecessary white space.

File: src/main/java/net/glowstone/GlowChunkSnapshot.java [hunk 8]
```diff
@@ -193,7 +181,7 @@ public int getBlockEmittedLight(int x, int y, int z) {
         public int getHighestBlockYAt(int x, int z) {
             return 0;
         }
-
+
```
Comment (Wild1145) [inline src/main/java/net/glowstone/GlowChunkSnapshot.java:184]: Unnecessary white space.

File: src/main/java/net/glowstone/GlowChunkSnapshot.java [hunk 9]
```diff
@@ -193,7 +181,7 @@ public int getBlockEmittedLight(int x, int y, int z) {
         public int getHighestBlockYAt(int x, int z) {
             return 0;
         }
-
+        
     }
-
+
```
Comment (Wild1145) [inline src/main/java/net/glowstone/GlowChunkSnapshot.java:186]: Unnecessary white space.

File: src/main/java/net/glowstone/GlowServer.java [hunk 10]
```diff
@@ -15,8 +14,8 @@
 import net.glowstone.scheduler.GlowScheduler;
 import net.glowstone.scheduler.WorldScheduler;
 import net.glowstone.util.*;
-import net.glowstone.util.bans.GlowBanList;
 import net.glowstone.util.bans.UuidListFile;
+import net.glowstone.util.bans.GlowBanList;
```
Comment (Wild1145) [inline src/main/java/net/glowstone/GlowServer.java:18]: Why? You removed the import just to add it in the next line?

**Comment by Tonodus** [comment]
This PR contains the commits made by others, so it looks like he want's to make the changes, but they're already done.

**Comment by Wild1145** [comment]
In src/main/java/net/glowstone/block/ItemTable.java you have removed a lot of blocks, again, it seems like a mistake?

**Comment by Aaron1011** [comment]
@Wild1145: I think that's just because the PR needs to be rebased against `master`.

**Comment by Wild1145** [comment]
@Aaron1011 Il be honest, this PR just confuses me... Its not able to be pulled anyway, but it just seems like someone has selected a random commit to pull and then tried merging and committing and made a bit of a mess of it...

**Comment by dequis** [comment]
Here's a rebased version with a much clearer diff:

https://github.com/dequis/Glowstone/compare/GlowstoneMC:master...dequis:pr-191

**Comment by Wild1145** [comment]
Now thats easier to read :)

**Pull request closed without merging**